src/api/route-services/work-response.js

```diff
@@ -7,7 +7,17 @@ const { handlePagination } = require('../../utils/handlePagination');
 const ExperimentService = require('./experiment');
 
 const NEW = 'NEW';
-const DATA_UPDATE = 'data_update';
+const WORKER_DATA_UPDATE = 'workerDataUpdate';
+
+const persistUpdates = async (experimentId, responseForClient) => {
+  const taskName = responseForClient.request.body.name;
+
+  if (taskName === 'ClusterCells') {
+    const cellSets = JSON.parse(responseForClient.results[0].body);
+
+    await (new ExperimentService()).updateLouvainCellSets(experimentId, cellSets);
```

Comment:
in the future, when we have more than one clustering method we will have to refactor this into a more generic update right?

Reply:
Yes, definitely. Also when we do that we will benefit from storing each cellset in a separate entry so that we don't have to do 2 operations to update an entry (getting current entry, merging it iwth the new cellset and putting it back)